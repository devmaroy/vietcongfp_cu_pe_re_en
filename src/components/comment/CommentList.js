import React from 'react';
import PropTypes from 'prop-types';
import Pagination from "react-js-pagination";
import CommentItem from './CommentItem';

class CommentList extends React.Component {
    state = {
        activePage: 1,
        commentsPerPage: 5
    }

    handlePageChange = ( pageNumber ) => {
        this.setState( () => ( { activePage: pageNumber } ) );
    }

    render() {
        const { activePage, commentsPerPage } = this.state;
        const { comments } = this.props;

        return (
            <div>
                <h3>Comments: { comments.length }</h3>
                {
                    comments.length === 0 ? (
                        <p>There are no comments.</p>
                    ) : (
                        <div>
                            <div className="comments">
                                {
                                    comments
                                        .slice( ( activePage - 1 ) * commentsPerPage, ( commentsPerPage * activePage ) )
                                        .map( ( comment ) => <CommentItem key={ comment.id } comment={ comment } /> )
                                }
                            </div>  
                            {  
                                comments.length > commentsPerPage && (
                                    <div className="site-pagination">
                                        <Pagination
                                            activePage={ this.state.activePage }
                                            itemsCountPerPage={ commentsPerPage }
                                            totalItemsCount={ comments.length }
                                            pageRangeDisplayed={ 5 }
                                            onChange={ this.handlePageChange }
                                        />
                                    </div>
                                )
                            }
                        </div>
                    )
                }
            </div>
        );
    };
};

CommentList.propTypes = {
    comments: PropTypes.array.isRequired
};

export default CommentList;