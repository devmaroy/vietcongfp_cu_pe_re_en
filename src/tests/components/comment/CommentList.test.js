import React from 'react';
import { shallow } from 'enzyme';
import CommentList from '../../../components/comment/CommentList';
import comments from '../../fixtures/comments';

test( 'should render CommentList with empty message', () => {
    const wrapper = shallow(
        <CommentList
            comments={ [] }
        />
    );

    expect( wrapper ).toMatchSnapshot();
});

test( 'should render CommentList correctly with data', () => {
    const wrapper = shallow(
        <CommentList
            comments={ comments }
        />  
    );

    expect( wrapper ).toMatchSnapshot();
});

test( 'should show Pagination if length of comments is more than commentsPerPage in state', () => {
    const wrapper = shallow(
        <CommentList
            comments={ comments }
        />  
    );

    wrapper.setState({
        commentsPerPage: 1
    });

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find( '.site-pagination' ).exists() ).toBeTruthy();
});

test( 'should not show Pagination if length of comments is less than commentsPerPage in state', () => {
    const wrapper = shallow(
        <CommentList
            comments={ comments }
        />  
    );

    wrapper.setState({
        commentsPerPage: 20
    });

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find( '.site-pagination' ).exists() ).toBeFalsy();
});