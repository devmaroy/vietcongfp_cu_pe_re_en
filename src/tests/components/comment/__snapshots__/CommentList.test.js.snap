// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not show Pagination if length of comments is less than commentsPerPage in state 1`] = `
<div>
  <h3>
    Comments: 
    5
  </h3>
  <div>
    <div
      className="comments"
    >
      <Connect(CommentItem)
        comment={
          Object {
            "body": "Oh. Awesome!!!",
            "createdAt": 1000,
            "id": "1",
            "name": "Andy",
            "postId": "1",
          }
        }
        key="1"
      />
      <Connect(CommentItem)
        comment={
          Object {
            "body": "Great! New patch is here!",
            "createdAt": 1200,
            "id": "2",
            "name": "Pete0009",
            "postId": "1",
          }
        }
        key="2"
      />
      <Connect(CommentItem)
        comment={
          Object {
            "body": "Nice pictures :))",
            "createdAt": 1500,
            "id": "3",
            "name": "Grer04",
            "postId": "2",
          }
        }
        key="3"
      />
      <Connect(CommentItem)
        comment={
          Object {
            "body": "I want to join your team!!! Please!",
            "createdAt": 2000,
            "id": "4",
            "name": "Iwane01",
            "postId": "3",
          }
        }
        key="4"
      />
      <Connect(CommentItem)
        comment={
          Object {
            "body": "Yeah.. of course.. we need more team members. MEH",
            "createdAt": 2200,
            "id": "5",
            "name": "Susi",
            "postId": "3",
          }
        }
        key="5"
      />
    </div>
  </div>
</div>
`;

exports[`should render CommentList correctly with data 1`] = `
<div>
  <h3>
    Comments: 
    5
  </h3>
  <div>
    <div
      className="comments"
    >
      <Connect(CommentItem)
        comment={
          Object {
            "body": "Oh. Awesome!!!",
            "createdAt": 1000,
            "id": "1",
            "name": "Andy",
            "postId": "1",
          }
        }
        key="1"
      />
      <Connect(CommentItem)
        comment={
          Object {
            "body": "Great! New patch is here!",
            "createdAt": 1200,
            "id": "2",
            "name": "Pete0009",
            "postId": "1",
          }
        }
        key="2"
      />
      <Connect(CommentItem)
        comment={
          Object {
            "body": "Nice pictures :))",
            "createdAt": 1500,
            "id": "3",
            "name": "Grer04",
            "postId": "2",
          }
        }
        key="3"
      />
      <Connect(CommentItem)
        comment={
          Object {
            "body": "I want to join your team!!! Please!",
            "createdAt": 2000,
            "id": "4",
            "name": "Iwane01",
            "postId": "3",
          }
        }
        key="4"
      />
      <Connect(CommentItem)
        comment={
          Object {
            "body": "Yeah.. of course.. we need more team members. MEH",
            "createdAt": 2200,
            "id": "5",
            "name": "Susi",
            "postId": "3",
          }
        }
        key="5"
      />
    </div>
  </div>
</div>
`;

exports[`should render CommentList with empty message 1`] = `
<div>
  <h3>
    Comments: 
    0
  </h3>
  <p>
    There are no comments.
  </p>
</div>
`;

exports[`should show Pagination if length of comments is more than commentsPerPage in state 1`] = `
<div>
  <h3>
    Comments: 
    5
  </h3>
  <div>
    <div
      className="comments"
    >
      <Connect(CommentItem)
        comment={
          Object {
            "body": "Oh. Awesome!!!",
            "createdAt": 1000,
            "id": "1",
            "name": "Andy",
            "postId": "1",
          }
        }
        key="1"
      />
    </div>
    <div
      className="site-pagination"
    >
      <a
        activePage={1}
        firstPageText="«"
        getPageUrl={[Function]}
        hideFirstLastPages={false}
        innerClass="pagination"
        itemsCountPerPage={1}
        lastPageText="»"
        nextPageText="⟩"
        onChange={[Function]}
        pageRangeDisplayed={5}
        prevPageText="⟨"
        totalItemsCount={5}
      />
    </div>
  </div>
</div>
`;
